### PR TITLE
allow Her::Model to be serialized and deserialized properly using YAML

### PR DESCRIPTION
**Pull request by samandmoore**
@jmileham @ceslami @reustmd

### CODE REVIEW

File: lib/her/model/attributes.rb [hunk 1]
```diff
@@ -87,7 +87,7 @@ def method_missing(method, *args, &blk)
 
       # @private
       def respond_to_missing?(method, include_private = false)
-        method.to_s.end_with?('=') || method.to_s.end_with?('?') || @attributes.include?(method) || super
+        method.to_s.end_with?('=') || method.to_s.end_with?('?') || attributes.include?(method) || super
```
Comment (ceslami) [inline lib/her/model/attributes.rb]: Mind coloring this a bit for me?

Comment (jmileham) [inline lib/her/model/attributes.rb]: YAML deserialization sidesteps `new`, and hence `initialize`, which means that @attributes is `nil` during the deserialization process.  That means that a her model's `#respond_to_missing?` will raise an exception during deserialization.  This uses the standard getter for `attributes` instead, which lazily stands up a HashWithIndifferentAccess, which causes `#respond_to_missing?` to behave itself in a pre-initialized state, and therefor the rest of the YAML deserialization process works like a charm.

Comment (ceslami) [inline lib/her/model/attributes.rb]: :sexy:

**Comment by ceslami** [comment]
LGTM

**Comment by reustmd** [comment]
lgtm

**Pull request merged**